From 72c598633b9c784ecf2eb49452bab3940d2ad575 Mon Sep 17 00:00:00 2001
From: Kay Potthoff <Kay.Potthoff@microsys.de>
Date: Wed, 17 Mar 2021 10:31:22 +0100
Subject: [PATCH 11/19] net: stmicro: check for MDIO bus
Organization: MicroSys Electronics GmbH

Added check not to return with error in case that MDIO bus
is not configured.

Signed-off-by: Kay Potthoff <Kay.Potthoff@microsys.de>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
index 40c42637ad75..7317676fe69f 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
@@ -357,11 +357,13 @@ int stmmac_mdio_register(struct net_device *ndev)
 	new_bus->phy_mask = mdio_bus_data->phy_mask;
 	new_bus->parent = priv->device;

+	if (of_device_is_available(mdio_node)) {
 	    err = of_mdiobus_register(new_bus, mdio_node);
 	    if (err != 0) {
 	        dev_err(dev, "Cannot register the MDIO bus\n");
 	        goto bus_register_fail;
 	    }
+	}

 	if (priv->plat->phy_node || mdio_node)
 		goto bus_register_done;
--
2.30.2

